tools/runners/run-mongo.js

```diff
@@ -607,38 +607,43 @@ var launchMongo = function (options) {
         });
       }
 
-      var initiateResult = yieldingMethod(
-        db.admin(), 'command', {replSetInitiate: configuration});
+      try{
+        var initiateResult = yieldingMethod(
+          db.admin(), 'command', {replSetInitiate: configuration});
+      } catch (e){
+        if(e.message !== 'already initialized') {
+          throw Error("rs.initiate error: " + e.message);
+        }
+      }
+
       if (stopped) {
         return;
       }
-      // why this isn't in the error is unclear.
-      if (initiateResult && initiateResult.documents
-          && initiateResult.documents[0]
-          && initiateResult.documents[0].errmsg) {
-        var err = initiateResult.documents[0].errmsg;
-        if (err !== "already initialized") {
-          throw Error("rs.initiate error: " +
-                      initiateResult.documents[0].errmsg);
-        }
-      }
+
       // XXX timeout eventually?
       while (!stopped) {
-        var status = yieldingMethod(db.admin(), 'command',
-                                    {replSetGetStatus: 1});
-        if (!(status && status.documents && status.documents[0])) {
-          throw status;
+        var status;
+        try{
+          status = yieldingMethod(db.admin(), 'command',
+                                      {replSetGetStatus: 1});
+        } catch (e) {
+          // Some of the expected results, come as errors.
+          status = e;
         }
-        status = status.documents[0];
+
+        // See https://docs.mongodb.com/manual/reference/replica-states/
+        // for information on various states
         if (!status.ok) {
-          if (status.startupStatus === 6) {  // "SOON"
+          // UNKNOWN(6) some of the other replicas are in an UNKNOWN state from
+          // this replica perspective.
+          if (status.myState === 6) {
```

Comment:
That is good you pointed me to that test. There is also some problems in the test itself regarding the driver changes.

I confess I was a little shy of running self-tests. No longer!! I have seen it happening once or twice while I was developing, but I will make sure that the mongo tests stress this one code branch or that I add a test that does (or remove it, but I think its necessary).